### DISCUSSION TITLE
Reset query error state when calling refetch

### DISCUSSION THREAD

**Discussion by vlanemcev** [TanStack/query#8207] [Q&A]
I remember that in version 4 of React Query the refreshing functionality worked as follows:

- I have a query that failed the first time I tried to download it. Accordingly, it is in the `isError: true` state
- if I then called `refetch` of this query, it was again assigned the `isLoading` state, reset error state and I could display, for example, the loader again instead of displaying the error. 

As far as I can see now, this behavior has changed and now the `refetch` call will not reset the query’s state to “initial loading” and will not reset the error for this query if it sees that the query does not have data. 

Correct me - is this true?

And if so, then what would be the best way to handle such a case?

I use QueryErrorResetBoundary with ErrorBoundary and put function in query where I check if this is first loading - then I just throw error to the boundary. In my boundary I have button to reload again.
But because of React Native - I want to give the user possibility to refetch errored query by "Pull to refresh" functionality.

**Comment by vlanemcev** [comment]
https://stackblitz.com/edit/tanstack-query-hnpf3s?file=src%2Findex.jsx

Okay, as I checked in this example with latest versions of tanstack query - everything working as expected about resetting error when `refetch` is calling (for example from Dev tools). 

But when we handle errors using QueryErrorBoudaryReset, our error will still remain when we call `refetch` in Dev tools and will reset when we click on button Reload. 

Is this expected behavior or I can create issue? 

@TkDodo

**Reply by TkDodo** [reply]
I don’t understand your reproduction, show some steps to reproduce please. For me:

initially, it shows:

```
isPending: true
isFetching: true
isError: false
error: undefined
```

then I wait 10seconds, and I see:

```
Error when initial loading
Reload now
```

I click `reload now`, and I see:

```
isPending: true
isFetching: true
isError: false
error: undefined
```

then I see again

```
Error when initial loading
Reload now
```

This is all very much expected. So what’s the issue ?
